client/signup/steps/user/index.jsx

```diff
@@ -115,6 +115,8 @@ export class UserStep extends Component {
 			locale: this.props.locale,
 			oauth2ClientId: this.props.oauth2Client && this.props.oauth2Client.id,
 			wccomFrom: this.props.wccomFrom,
+			isReskinned: this.props.isReskinned,
+			signupUrl: window.location.href,
```

Comment:
The purpose of this is that the "create a new account" link on the login page navigates back to the `user` step, rather than to the start of the signup flow?

Reply:
I believe that is correct. Looks like Gutenboarding creates the signup URL to be whatever's defined [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/login/index.js#L105). This happens on any visit to `/log-in/new` (without the `signup_url` param in this PR), so also from coming back from the magic link page. 

That said, I feel we may end up reverting this logic (last two commits) altogether if the scenario described [here](https://github.com/Automattic/wp-calypso/pull/54646#issuecomment-886685677) is of any concern. If `/log-in/new?signup_url=http://foo.bar` forces `http://foo.bar` for a "create a new account" link, then that may be something we'd want to avoid?

Thanks for the review! 🙇

Reply:
> In this case we could accept only relative signup URLs, because the site is always https://wordpress.com

This feels like the easiest way forward for now. Addressed in https://github.com/Automattic/wp-calypso/pull/54646/commits/ac88547eeffe955057c3adffee4188167e2b5757